packages/roosterjs-editor-core/lib/coreAPI/insertNode.ts

```diff
@@ -71,10 +71,14 @@ export const insertNode: InsertNode = (core: EditorCore, node: Node, option: Ins
                     // For insert on new line, or refNode is text or void html element (HR, BR etc.)
                     // which cannot have children, i.e. <div>hello<br>world</div>. 'hello', 'world' are the
                     // first and last node. Insert before 'hello' or after 'world', but still inside DIV
-                    insertedNode = refNode.parentNode.insertBefore(
-                        node,
-                        isBegin ? refNode : refNode.nextSibling
-                    );
+                    if (isBegin || !option.insertOnNewLine) {
+                        insertedNode = refNode.parentNode.insertBefore(
+                            node,
+                            isBegin ? refNode : refNode.nextSibling
+                        );
+                    } else {
+                        contentDiv.appendChild(node);
```

Comment:
Do we need to ensure `insertedNode` is set to that it can be wrapped in a div if necessary?

Reply:
Good point - updated